build/xtask/src/dist.rs

```diff
@@ -2524,28 +2519,74 @@ fn write_elf(
     // Page align the start of sections data.
     let sections_data_offset = (program_headers_offset + 0xfff) & !0xfff;
 
+    // Page align the Section Header String Table.
+    let shstrtab_offset =
+        (sections_data_offset + sections_data.len() + 0xfff) & !0xfff;
+
+    // Create a Section Header String Table, to hold the actual section
+    // names.
```

Comment:
Would the section names hold the task names?

Reply:
I don't think we have access to that during this point of packaging - but we _can_ use the source filename, which I believe includes the end 'elf' (which is typically a 1:1 mapping between source elf and task, and a many:1 relationship between sections and source elf).

Reply:
Sounds good. We can also do this as a follow up.